### PR TITLE
Update project to recommended settings

### PR DESCRIPTION
**Pull request by tiagomartinho**
1. Update project to recommended settings:
~~- Disable Code Signing for Target 'CareKit'~~
- Enable Recommended compiler Warnings

2. Set Swift version to 3.0 in OCKTest-iPad Project
 
3. Explicitly cast error description String to Any

4. Remove Swift compiler directive in OCKSample

5. Remove Swift compiler directive in OCKTest

### CODE REVIEW

File: CareKit.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -1162,6 +1167,7 @@
 		8605A5D01C4F04EC00DD65FF /* Release */ = {
 			isa = XCBuildConfiguration;
 			buildSettings = {
+				"CODE_SIGN_IDENTITY[sdk=iphoneos*]" = "";
```
Comment (umerkhan-apple) [inline CareKit.xcodeproj/project.pbxproj]: Do we need this?

Comment (tiagomartinho) [inline CareKit.xcodeproj/project.pbxproj]: It was suggested by Xcode when updating the project to the recommended settings.

I can remove it if you think we don't need it.

If it should be removed, should it be removed both for the Debug and Release build configurations?

Comment (umerkhan-apple) [inline CareKit.xcodeproj/project.pbxproj]: Yes, from both configurations. Thanks.

**Comment by tiagomartinho** [comment]
Done!

**Comment by umerkhan-apple** [comment]
@tiagomartinho Thanks!

**Comment by tiagomartinho** [comment]
Thank you!

**Pull request merged**